data/core/commandview.lua

```diff
@@ -274,18 +275,31 @@ function CommandView:get_suggestion_line_height()
 end
 
 
-function CommandView:update_suggestions()
-  local t = self.state.suggest(self:get_text()) or {}
+function CommandView:update_suggestions(preserve)
+  local text = self:get_text()
+  local t = self.state.suggest(text == self.save_suggestion and "" or text) or {}
   local res = {}
   for i, item in ipairs(t) do
     if type(item) == "string" then
       item = { text = item }
     end
     res[i] = item
   end
+  if preserve and self.suggestions and self.suggestion_idx ~= 1 then
```

Comment:
Does this mean that if the first suggestion is selected, the selection moves around?

Reply:
Yes; it's basically saying that if you haven't modified your selection basically, let it move around.

Reply:
I mean, one could go back to the first one.

Reply:
True, so you think we should count as preserving it if the last action you took was up/down, and reset that if you start typing something?

Reply:
Yeah, makes sense.

Reply:
Done.